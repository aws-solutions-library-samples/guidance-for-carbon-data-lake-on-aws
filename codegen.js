"use strict";
// Description: Codegen configuration file for GraphQL Code Generator
// @see https://graphql-code-generator.com/docs/getting-started/codegen-config
Object.defineProperty(exports, "__esModule", { value: true });
const schemaLocation = ['lib/stacks/stack-api/schema.graphql', 'lib/stacks/stack-api/appsync.graphql'];
// For GLEC compliance copy content from ./configuration-examples/GLEC/activity-schema.graphql into ./activity-schema.graphql file
const config = {
    emitLegacyCommonJSImports: true,
    schema: schemaLocation,
    generates: {
        './lib/stacks/stack-web/app/sample-ui-cloudscape/public/assets/activity-input-template.csv': {
            plugins: ['./lib/codegen/csv.ts'],
            config: {
                targetType: 'Activity',
            },
        },
        // TODO: Graphql javascript operations generator
        // 'lib/stacks/stack-web/app/sample-ui-cloudscape/src/graphql/types.ts': {
        //   schema: ['lib/stacks/stack-api/schema.graphql', 'lib/stacks/stack-api/appsync.graphql'],
        //   plugins: ['typescript', 'typescript-operations'],
        //   config: {
        //     scalars: {
        //       AWSJSON: 'string',
        //       AWSDate: 'string',
        //       AWSTime: 'string',
        //       AWSDateTime: 'string',
        //       AWSTimestamp: 'number',
        //       AWSEmail: 'string',
        //       AWSURL: 'string',
        //       AWSPhone: 'string',
        //       AWSIPAddress: 'string',
        //     },
        //   },
        // },
        // TODO: DQ generator
        // lib/stacks/stack-data-pipeline/construct-data-quality/lambda/manage_dq_resources/dq_rules.json': {
        //   schema: schemaLocation,
        //   plugins: [__dirname + './lib/codegen/dq-rules.ts'],
        // },
        // TODO: Python types generator for calculator
        // 'lib/stacks/stack-data-pipeline/construct-calculator/lambda/types.py': {
        //   schema: schemaLocation,
        //   plugins: [__dirname + './lib/codegen/python-types.ts'],
        // },
    },
};
exports.default = config;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiY29kZWdlbi5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbImNvZGVnZW4udHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IjtBQUFBLHFFQUFxRTtBQUNyRSw4RUFBOEU7O0FBSTlFLE1BQU0sY0FBYyxHQUFHLENBQUMscUNBQXFDLEVBQUUsc0NBQXNDLENBQUMsQ0FBQTtBQUV0RyxrSUFBa0k7QUFFbEksTUFBTSxNQUFNLEdBQWtCO0lBQzVCLHlCQUF5QixFQUFFLElBQUk7SUFDL0IsTUFBTSxFQUFFLGNBQWM7SUFDdEIsU0FBUyxFQUFFO1FBQ1QsMkZBQTJGLEVBQUU7WUFDM0YsT0FBTyxFQUFFLENBQUMsc0JBQXNCLENBQUM7WUFDakMsTUFBTSxFQUFFO2dCQUNOLFVBQVUsRUFBRSxVQUFVO2FBQ3ZCO1NBQ0Y7UUFDRCxnREFBZ0Q7UUFDaEQsMEVBQTBFO1FBQzFFLDZGQUE2RjtRQUM3RixzREFBc0Q7UUFDdEQsY0FBYztRQUNkLGlCQUFpQjtRQUNqQiwyQkFBMkI7UUFDM0IsMkJBQTJCO1FBQzNCLDJCQUEyQjtRQUMzQiwrQkFBK0I7UUFDL0IsZ0NBQWdDO1FBQ2hDLDRCQUE0QjtRQUM1QiwwQkFBMEI7UUFDMUIsNEJBQTRCO1FBQzVCLGdDQUFnQztRQUNoQyxTQUFTO1FBQ1QsT0FBTztRQUNQLEtBQUs7UUFDTCxxQkFBcUI7UUFDckIscUdBQXFHO1FBQ3JHLDRCQUE0QjtRQUM1Qix3REFBd0Q7UUFDeEQsS0FBSztRQUNMLDhDQUE4QztRQUM5QywyRUFBMkU7UUFDM0UsNEJBQTRCO1FBQzVCLDREQUE0RDtRQUM1RCxLQUFLO0tBQ047Q0FDRixDQUFBO0FBRUQsa0JBQWUsTUFBTSxDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiLy8gRGVzY3JpcHRpb246IENvZGVnZW4gY29uZmlndXJhdGlvbiBmaWxlIGZvciBHcmFwaFFMIENvZGUgR2VuZXJhdG9yXG4vLyBAc2VlIGh0dHBzOi8vZ3JhcGhxbC1jb2RlLWdlbmVyYXRvci5jb20vZG9jcy9nZXR0aW5nLXN0YXJ0ZWQvY29kZWdlbi1jb25maWdcblxuaW1wb3J0IHsgQ29kZWdlbkNvbmZpZyB9IGZyb20gJ0BncmFwaHFsLWNvZGVnZW4vY2xpJ1xuXG5jb25zdCBzY2hlbWFMb2NhdGlvbiA9IFsnbGliL3N0YWNrcy9zdGFjay1hcGkvc2NoZW1hLmdyYXBocWwnLCAnbGliL3N0YWNrcy9zdGFjay1hcGkvYXBwc3luYy5ncmFwaHFsJ11cblxuLy8gRm9yIEdMRUMgY29tcGxpYW5jZSBjb3B5IGNvbnRlbnQgZnJvbSAuL2NvbmZpZ3VyYXRpb24tZXhhbXBsZXMvR0xFQy9hY3Rpdml0eS1zY2hlbWEuZ3JhcGhxbCBpbnRvIC4vYWN0aXZpdHktc2NoZW1hLmdyYXBocWwgZmlsZVxuXG5jb25zdCBjb25maWc6IENvZGVnZW5Db25maWcgPSB7XG4gIGVtaXRMZWdhY3lDb21tb25KU0ltcG9ydHM6IHRydWUsXG4gIHNjaGVtYTogc2NoZW1hTG9jYXRpb24sXG4gIGdlbmVyYXRlczoge1xuICAgICcuL2xpYi9zdGFja3Mvc3RhY2std2ViL2FwcC9zYW1wbGUtdWktY2xvdWRzY2FwZS9wdWJsaWMvYXNzZXRzL2FjdGl2aXR5LWlucHV0LXRlbXBsYXRlLmNzdic6IHtcbiAgICAgIHBsdWdpbnM6IFsnLi9saWIvY29kZWdlbi9jc3YudHMnXSxcbiAgICAgIGNvbmZpZzoge1xuICAgICAgICB0YXJnZXRUeXBlOiAnQWN0aXZpdHknLFxuICAgICAgfSxcbiAgICB9LFxuICAgIC8vIFRPRE86IEdyYXBocWwgamF2YXNjcmlwdCBvcGVyYXRpb25zIGdlbmVyYXRvclxuICAgIC8vICdsaWIvc3RhY2tzL3N0YWNrLXdlYi9hcHAvc2FtcGxlLXVpLWNsb3Vkc2NhcGUvc3JjL2dyYXBocWwvdHlwZXMudHMnOiB7XG4gICAgLy8gICBzY2hlbWE6IFsnbGliL3N0YWNrcy9zdGFjay1hcGkvc2NoZW1hLmdyYXBocWwnLCAnbGliL3N0YWNrcy9zdGFjay1hcGkvYXBwc3luYy5ncmFwaHFsJ10sXG4gICAgLy8gICBwbHVnaW5zOiBbJ3R5cGVzY3JpcHQnLCAndHlwZXNjcmlwdC1vcGVyYXRpb25zJ10sXG4gICAgLy8gICBjb25maWc6IHtcbiAgICAvLyAgICAgc2NhbGFyczoge1xuICAgIC8vICAgICAgIEFXU0pTT046ICdzdHJpbmcnLFxuICAgIC8vICAgICAgIEFXU0RhdGU6ICdzdHJpbmcnLFxuICAgIC8vICAgICAgIEFXU1RpbWU6ICdzdHJpbmcnLFxuICAgIC8vICAgICAgIEFXU0RhdGVUaW1lOiAnc3RyaW5nJyxcbiAgICAvLyAgICAgICBBV1NUaW1lc3RhbXA6ICdudW1iZXInLFxuICAgIC8vICAgICAgIEFXU0VtYWlsOiAnc3RyaW5nJyxcbiAgICAvLyAgICAgICBBV1NVUkw6ICdzdHJpbmcnLFxuICAgIC8vICAgICAgIEFXU1Bob25lOiAnc3RyaW5nJyxcbiAgICAvLyAgICAgICBBV1NJUEFkZHJlc3M6ICdzdHJpbmcnLFxuICAgIC8vICAgICB9LFxuICAgIC8vICAgfSxcbiAgICAvLyB9LFxuICAgIC8vIFRPRE86IERRIGdlbmVyYXRvclxuICAgIC8vIGxpYi9zdGFja3Mvc3RhY2stZGF0YS1waXBlbGluZS9jb25zdHJ1Y3QtZGF0YS1xdWFsaXR5L2xhbWJkYS9tYW5hZ2VfZHFfcmVzb3VyY2VzL2RxX3J1bGVzLmpzb24nOiB7XG4gICAgLy8gICBzY2hlbWE6IHNjaGVtYUxvY2F0aW9uLFxuICAgIC8vICAgcGx1Z2luczogW19fZGlybmFtZSArICcuL2xpYi9jb2RlZ2VuL2RxLXJ1bGVzLnRzJ10sXG4gICAgLy8gfSxcbiAgICAvLyBUT0RPOiBQeXRob24gdHlwZXMgZ2VuZXJhdG9yIGZvciBjYWxjdWxhdG9yXG4gICAgLy8gJ2xpYi9zdGFja3Mvc3RhY2stZGF0YS1waXBlbGluZS9jb25zdHJ1Y3QtY2FsY3VsYXRvci9sYW1iZGEvdHlwZXMucHknOiB7XG4gICAgLy8gICBzY2hlbWE6IHNjaGVtYUxvY2F0aW9uLFxuICAgIC8vICAgcGx1Z2luczogW19fZGlybmFtZSArICcuL2xpYi9jb2RlZ2VuL3B5dGhvbi10eXBlcy50cyddLFxuICAgIC8vIH0sXG4gIH0sXG59XG5cbmV4cG9ydCBkZWZhdWx0IGNvbmZpZ1xuIl19