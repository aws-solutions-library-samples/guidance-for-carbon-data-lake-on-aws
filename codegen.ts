// Description: Codegen configuration file for GraphQL Code Generator
// @see https://graphql-code-generator.com/docs/getting-started/codegen-config

import { CodegenConfig } from '@graphql-codegen/cli'

const schemaLocation = ['lib/stacks/stack-api/schema.graphql', 'lib/stacks/stack-api/appsync.graphql']

// For GLEC compliance copy content from ./configuration-examples/GLEC/activity-schema.graphql into ./activity-schema.graphql file

const config: CodegenConfig = {
  emitLegacyCommonJSImports: true,
  schema: schemaLocation,
  generates: {
    './lib/stacks/stack-web/app/sample-ui-cloudscape/public/assets/activity-input-template.csv': {
      plugins: ['./lib/codegen/csv.ts'],
      config: {
        targetType: 'Activity',
      },
    },
    // TODO: Graphql javascript operations generator
    // 'lib/stacks/stack-web/app/sample-ui-cloudscape/src/graphql/types.ts': {
    //   schema: ['lib/stacks/stack-api/schema.graphql', 'lib/stacks/stack-api/appsync.graphql'],
    //   plugins: ['typescript', 'typescript-operations'],
    //   config: {
    //     scalars: {
    //       AWSJSON: 'string',
    //       AWSDate: 'string',
    //       AWSTime: 'string',
    //       AWSDateTime: 'string',
    //       AWSTimestamp: 'number',
    //       AWSEmail: 'string',
    //       AWSURL: 'string',
    //       AWSPhone: 'string',
    //       AWSIPAddress: 'string',
    //     },
    //   },
    // },
    // TODO: DQ generator
    // lib/stacks/stack-data-pipeline/construct-data-quality/lambda/manage_dq_resources/dq_rules.json': {
    //   schema: schemaLocation,
    //   plugins: [__dirname + './lib/codegen/dq-rules.ts'],
    // },
    // TODO: Python types generator for calculator
    // 'lib/stacks/stack-data-pipeline/construct-calculator/lambda/types.py': {
    //   schema: schemaLocation,
    //   plugins: [__dirname + './lib/codegen/python-types.ts'],
    // },
  },
}

export default config
