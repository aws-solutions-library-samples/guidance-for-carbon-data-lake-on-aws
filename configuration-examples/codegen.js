"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const codegen_1 = require("../lib/codegen");
const config = {
    emitLegacyCommonJSImports: false,
    generates: {
        './__generated__/activity.csv': {
            schema: './activity-schema.graphql',
            plugins: ['../lib/index.js'],
            config: {
                fileFormat: codegen_1.FileFormat.CSV,
            },
        },
    },
};
exports.default = config;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiY29kZWdlbi5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbImNvZGVnZW4udHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7QUFDQSw0Q0FBNEM7QUFFNUMsTUFBTSxNQUFNLEdBQWtCO0lBQzVCLHlCQUF5QixFQUFFLEtBQUs7SUFDaEMsU0FBUyxFQUFFO1FBQ1QsOEJBQThCLEVBQUU7WUFDOUIsTUFBTSxFQUFFLDJCQUEyQjtZQUNuQyxPQUFPLEVBQUUsQ0FBQyxpQkFBaUIsQ0FBQztZQUM1QixNQUFNLEVBQUU7Z0JBQ04sVUFBVSxFQUFFLG9CQUFVLENBQUMsR0FBRzthQUMzQjtTQUNGO0tBQ0Y7Q0FDRixDQUFDO0FBRUYsa0JBQWUsTUFBTSxDQUFDIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgQ29kZWdlbkNvbmZpZyB9IGZyb20gJ0BncmFwaHFsLWNvZGVnZW4vY2xpJztcbmltcG9ydCB7IEZpbGVGb3JtYXQgfSBmcm9tICcuLi9saWIvY29kZWdlbic7XG5cbmNvbnN0IGNvbmZpZzogQ29kZWdlbkNvbmZpZyA9IHtcbiAgZW1pdExlZ2FjeUNvbW1vbkpTSW1wb3J0czogZmFsc2UsXG4gIGdlbmVyYXRlczoge1xuICAgICcuL19fZ2VuZXJhdGVkX18vYWN0aXZpdHkuY3N2Jzoge1xuICAgICAgc2NoZW1hOiAnLi9hY3Rpdml0eS1zY2hlbWEuZ3JhcGhxbCcsXG4gICAgICBwbHVnaW5zOiBbJy4uL2xpYi9pbmRleC5qcyddLFxuICAgICAgY29uZmlnOiB7XG4gICAgICAgIGZpbGVGb3JtYXQ6IEZpbGVGb3JtYXQuQ1NWLFxuICAgICAgfSxcbiAgICB9LFxuICB9LFxufTtcblxuZXhwb3J0IGRlZmF1bHQgY29uZmlnOyJdfQ==