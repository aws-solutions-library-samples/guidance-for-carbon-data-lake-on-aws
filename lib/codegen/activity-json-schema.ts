import { PluginFunction, Types } from '@graphql-codegen/plugin-helpers'
import { GraphQLSchema} from 'graphql'
import tmp from 'tmp';
import fs from 'fs';
import { plugin as typescriptPlugin } from '@graphql-codegen/typescript';
import * as tjs from "typescript-json-schema";
import { resolve } from "path";
import replaceInFile from 'replace-in-file'

export const plugin: PluginFunction = async (schema: GraphQLSchema, _documents: Types.DocumentFile[], _config: any) => {
  // Use graphql codegen typescript plugin to generate typscript types
  const types = await typescriptPlugin(schema, [], _config);

  // Write those typescript types out to a temp file
  const typesFile = writeSyncToTemporaryFile(types.content, '.ts');

  // Remove the Maybe type that is generated by the graphql typescript plugin as it causes
  // the json schema generation library to not be able to create the correct types
  await replaceInFile.replaceInFile({
    files: typesFile,
    from: new RegExp('(Maybe<.*>)', 'gm'),
    to: (match: string) => match.replace('Maybe<', '').replace('>', ''),
  })

  // Create the program to generate a json schema
  const program = tjs.getProgramFromFiles(
    [resolve(typesFile)]
  );

  const settings: tjs.PartialArgs = {
    required: true,
    ignoreErrors: true
  };

  // Generate json schema from typescript Activity type
  const jsonSchema = tjs.generateSchema(program, "Activity", settings);

  return JSON.stringify(jsonSchema, null, 2);
};

const writeSyncToTemporaryFile = (content: string, extension: string) => {
  const path = tmp.fileSync({ postfix: extension }).name;
  fs.writeFileSync(path, content);
  return path;
};
