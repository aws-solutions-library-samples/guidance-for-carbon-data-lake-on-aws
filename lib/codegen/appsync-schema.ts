import { ModelTransformer } from '@aws-amplify/graphql-model-transformer'
import { GraphQLTransform } from '@aws-amplify/graphql-transformer-core'
import { PluginFunction, Types } from '@graphql-codegen/plugin-helpers'
import { DirectiveNode, GraphQLDirective, GraphQLSchema, isSpecifiedDirective, isSpecifiedScalarType, print } from 'graphql'

export const plugin: PluginFunction = async (schema: GraphQLSchema, _documents: Types.DocumentFile[], _config: any) => {
  const modelTransformer = new ModelTransformer();
  const transformer = new GraphQLTransform({
    transformers: [modelTransformer],
  })

  const schemaAsString = printSchemaWithDirectives(schema, ['model'])

  const out = transformer.transform(schemaAsString)

  return out.schema
}

export const printSchemaWithDirectives = (schema: GraphQLSchema, includeDirectives: string[]): string => {
  const str = Object.keys(schema.getTypeMap())
    .filter(k => !k.match(/^__/))
    .reduce((accum: string, name: string) => {
      const type = schema.getType(name)
      if (
        type !== undefined &&
        type !== null &&
        !isSpecifiedScalarType(type as any) &&
        type?.astNode?.kind != 'ScalarTypeDefinition'
      ) {
        let printAST = print(type.astNode as any)
        // remove directives that are not in the includeDirectives list
        const directives: DirectiveNode[] = [];
        if(type.astNode?.directives) directives.push(...type.astNode.directives);
        if(type.astNode?.kind === 'ObjectTypeDefinition' && type.astNode?.fields) {
          type.astNode.fields.forEach(f => {if(f.directives) directives.push(...f.directives)});
        }
        console.log(`directives for ${name}: ${directives.map(d => d.name.value)}`)
        const toRemoveDirectives = directives.filter(d => !includeDirectives.includes(d?.name.value))
      
        if (toRemoveDirectives && toRemoveDirectives?.length > 0) {
          for (const directive of toRemoveDirectives) {
            console.log(`removing directive ${directive.name.value}`)
            const regex = new RegExp(`@${directive.name.value}`, 'g')
            printAST = printAST.replace(regex, '')
          }
        }

        accum += `${printAST}\n`
      }

      return accum
    }, '')

  return schema.getDirectives().reduce((accum: string, d: GraphQLDirective) => {
    return !isSpecifiedDirective(d) && d.astNode ? (accum += `${print(d.astNode)}\n`) : accum
  }, str + `\n`)
}
