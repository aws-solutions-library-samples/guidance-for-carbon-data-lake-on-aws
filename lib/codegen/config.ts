// Description: Codegen configuration file for GraphQL Code Generator
// @see https://graphql-code-generator.com/docs/getting-started/codegen-config

import { CodegenConfig } from '@graphql-codegen/cli'


// For GLEC compliance copy content from ./configuration-examples/GLEC/activity-schema.graphql into ./activity-schema.graphql file

export const config = (framework: string): CodegenConfig =>  {
  const schemaLocation = `./framework_configurations/${framework}/schema.graphql`;

  return {
  emitLegacyCommonJSImports: true,
  schema: schemaLocation,
  generates: {
    // Generate CSV input template for Activity input
    './lib/stacks/stack-web/app/sample-ui-cloudscape/public/assets/activity-input-template.csv': {
      plugins: ['./lib/codegen/csv.ts'],
      config: {
        targetType: 'Activity',
        onlyIncludeDirectives: ['input'],
      },
    },
    './lib/constructs/construct-rest-api/activity-schema.json': {
      plugins: ['./lib/codegen/activity-json-schema.ts']
    },
    // TODO: Graphql javascript operations generator
    // 'lib/stacks/stack-web/app/sample-ui-cloudscape/src/graphql/types.ts': {
    //   schema: ['lib/stacks/stack-api/schema.graphql', 'lib/stacks/stack-api/appsync.graphql'],
    //   plugins: ['typescript', 'typescript-operations'],
    //   config: {
    //     scalars: {
    //       AWSJSON: 'string',
    //       AWSDate: 'string',
    //       AWSTime: 'string',
    //       AWSDateTime: 'string',
    //       AWSTimestamp: 'number',
    //       AWSEmail: 'string',
    //       AWSURL: 'string',
    //       AWSPhone: 'string',
    //       AWSIPAddress: 'string',
    //     },
    //   },
    // },
    // Generate AppSync schema for the API
    './lib/stacks/stack-api/schema.graphql': {
      plugins: ['./lib/codegen/appsync-schema.ts'],
    },
    // Generate GraphQL operations for the UI
    './lib/stacks/stack-web/app/sample-ui-cloudscape/src/graphql/operations.js': {
      plugins: ['./lib/codegen/javascript-operations.ts'],
    },
    // TODO: DQ generator
    // lib/stacks/stack-data-pipeline/construct-data-quality/lambda/manage_dq_resources/dq_rules.json': {
    //   schema: schemaLocation,
    //   plugins: [__dirname + './lib/codegen/dq-rules.ts'],
    // },
    // Emission Factors keys
    './lib/stacks/stack-data-pipeline/construct-calculator/lambda/emissionFactorsLookupFields.json': {
      plugins: ['./lib/codegen/field-directives-to-json-array.ts'],
      config: {
        targetType: 'EmissionFactor',
        directive: 'lookup'
      },
    },
  },
}};

export default config
