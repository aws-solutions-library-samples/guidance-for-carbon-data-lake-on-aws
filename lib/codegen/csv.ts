import { PluginFunction, Types } from '@graphql-codegen/plugin-helpers';
import { GraphQLSchema } from 'graphql';
import { schemaToHeaders } from './utils';

export enum FileFormat {
  CSV,
  XSLX,
}

export type InputFileGeneratorConfig = {
  includeTypes: string[];
  fileFormat: FileFormat;
};

export const plugin: PluginFunction = async (
  schema: GraphQLSchema,
  _documents: Types.DocumentFile[],
  config: InputFileGeneratorConfig,
) => {
  const data = schemaToHeaders(schema);
  let result = '';

  if (config.fileFormat === FileFormat.XSLX) {
    // Not implemented
    throw new Error('Not implemented');
  }

  if (config.fileFormat === FileFormat.CSV) {
    for (const sheet of data) {
      for (const column of sheet.columns) {
        result += `${column.label.replace(/([A-Z])/g, '_$1').toLowerCase()},`;
      }
      result += '\n';
    }
  }

  return result;
};
