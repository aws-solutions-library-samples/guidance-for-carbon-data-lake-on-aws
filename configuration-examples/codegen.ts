import { CodegenConfig } from '@graphql-codegen/cli';
import { FileFormat } from '../lib/codegen';

const config: CodegenConfig = {
  emitLegacyCommonJSImports: false,
  generates: {
    './__generated__/activity.csv': {
      schema: './activity-schema.graphql',
      plugins: ['../lib/index.js'],
      config: {
        fileFormat: FileFormat.CSV,
      },
    },
  },
};

export default config;