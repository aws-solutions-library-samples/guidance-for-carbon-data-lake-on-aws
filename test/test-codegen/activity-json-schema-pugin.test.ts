import { buildSchema } from 'graphql'
import { plugin } from '../../lib/codegen/activity-json-schema'

describe('activity json schema generator', () => {
  it('Should output a json schema from a graphql type for Activity', async () => {
    // GIVEN
    const schema = buildSchema(
      /* GraphQL */ `
      scalar AWSDate
      scalar AWSTime
      scalar AWSDateTime
      scalar AWSTimestamp
      scalar AWSEmail
      scalar AWSJSON
      scalar AWSURL
      scalar AWSPhone
      scalar AWSIPAddress
      
      type Activity @model {
          id: ID #required
          activity_event_id: String! @input
          asset_id: String @input
          activity: String @input
          category: String @input
          scope: String @input
          geo: AWSJSON @input
          origin_measurement_timestamp: String @input
          raw_data: Float @input
          source: String @input
          units: String @input
          emissions_output: AWSJSON
          createdAt: AWSDateTime #required
          updatedAt: AWSDateTime #required
      }
      
      type EmissionFactor {
          category: String! @lookup
          activity: String! @lookup
          scope: Int! @lookup
          co2_factor: Float
          ch4_factor: Float
          n2o_factor: Float
          biofuel_co2: Float
          AR4_kgco2e: Float
          AR5_kgco2e: Float
          units: String!
          last_updated: String
          source: String
          source_origin: String
      }
      
      directive @lookup on FIELD_DEFINITION
      directive @input on FIELD_DEFINITION
      `,
      {
        assumeValidSDL: true,
      }
    )

    // WHEN
    const result = await plugin(schema, [], {})

    // THEN
    expect(result).toMatchSnapshot()
  })
})
