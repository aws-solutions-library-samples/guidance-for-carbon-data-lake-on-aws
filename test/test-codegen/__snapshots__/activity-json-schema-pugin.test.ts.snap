// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity json schema generator Should output a json schema from a graphql type for Activity 1`] = `
"{
  \\"type\\": \\"object\\",
  \\"properties\\": {
    \\"__typename\\": {
      \\"type\\": \\"string\\",
      \\"enum\\": [
        \\"Activity\\"
      ]
    },
    \\"id\\": {
      \\"type\\": \\"string\\"
    },
    \\"activity_event_id\\": {
      \\"type\\": \\"string\\"
    },
    \\"asset_id\\": {
      \\"type\\": \\"string\\"
    },
    \\"activity\\": {
      \\"type\\": \\"string\\"
    },
    \\"category\\": {
      \\"type\\": \\"string\\"
    },
    \\"scope\\": {
      \\"type\\": \\"string\\"
    },
    \\"geo\\": {},
    \\"origin_measurement_timestamp\\": {
      \\"type\\": \\"string\\"
    },
    \\"raw_data\\": {
      \\"type\\": \\"number\\"
    },
    \\"source\\": {
      \\"type\\": \\"string\\"
    },
    \\"units\\": {
      \\"type\\": \\"string\\"
    },
    \\"emissions_output\\": {},
    \\"createdAt\\": {},
    \\"updatedAt\\": {}
  },
  \\"required\\": [
    \\"activity_event_id\\"
  ],
  \\"$schema\\": \\"http://json-schema.org/draft-07/schema#\\"
}"
`;
